src/irc/core/irc-servers.h

```diff
@@ -69,6 +69,10 @@ struct _IRC_SERVER_REC {
 	int max_whois_in_cmd; /* max. number of nicks in one /WHOIS command */
 	int max_msgs_in_cmd; /* max. number of targets in one /MSG */
 
+	GSList *cap_supported; /* A list of caps supported by the server */
+	GSList *cap_active; /* A list of the caps active for this session */
```

Comment:
So, hm, this is a bit misleading, since you actually add sasl here before it's been acknowledged, and remove it on NAK. So I guess it's more like a list of requested-and-probably-active caps.